src/us/kbase/workspace/test/database/provenance/CreditMetadataTest.java

```diff
@@ -0,0 +1,883 @@
+					ELS, DATE_LIST, ELF, ELRI);
+		}
+
+		// overwrite .withResourceTypeString
+		for (Map.Entry<String, ResourceType> entry : VALID_RESOURCE_TYPE_MAP.entrySet()) {
+			final CreditMetadata creditMetadata = CreditMetadata
+					.getBuilder(IDENTIFIER_STRING, CONTRIBUTOR_LIST, TITLE_LIST)
+					.withDates(DATE_LIST)
+					.withResourceTypeString(entry.getKey())
+					.withResourceTypeString(null)
+					.build();
+			assertCreditMetadataFields(creditMetadata, IDENTIFIER_STRING, rtd,
+					CONTRIBUTOR_LIST, TITLE_LIST, EM,
+					ELS, DATE_LIST, ELF, ELRI);
+		}
+
+		// overwrite .withResourceType
+		for (final ResourceType rt : ResourceType.values()) {
+			final CreditMetadata creditMetadata = CreditMetadata
+					.getBuilder(IDENTIFIER_STRING, CONTRIBUTOR_LIST, TITLE_LIST)
+					.withDates(DATE_LIST)
+					.withResourceType(rt)
+					.withResourceTypeString(null)
+					.build();
+			assertCreditMetadataFields(creditMetadata, IDENTIFIER_STRING, rtd,
+					CONTRIBUTOR_LIST, TITLE_LIST, EM,
+					ELS, DATE_LIST, ELF, ELRI);
+		}
+
+		// overwrite .withResourceTypeString
+		for (Map.Entry<String, ResourceType> entry : VALID_RESOURCE_TYPE_MAP.entrySet()) {
+			final CreditMetadata creditMetadata = CreditMetadata
+					.getBuilder(IDENTIFIER_STRING, CONTRIBUTOR_LIST, TITLE_LIST)
+					.withDates(DATE_LIST)
+					.withResourceTypeString(entry.getKey())
+					.withResourceType(null)
+					.build();
+			assertCreditMetadataFields(creditMetadata, IDENTIFIER_STRING, rtd,
+					CONTRIBUTOR_LIST, TITLE_LIST, EM,
+					ELS, DATE_LIST, ELF, ELRI);
+		}
+	}
+
+	@Test
+	public void buildMaximalOverwriteNullEmptyLists() throws Exception {
+		for (final ResourceType rt : ResourceType.values()) {
+			final CreditMetadata creditMetadata = CreditMetadata
+					.getBuilder(IDENTIFIER_STRING, CONTRIBUTOR_LIST, TITLE_LIST)
+					.withResourceType(rt)
+					.withDates(DATE_LIST)
+					.withDates(Arrays.asList(null, null, null, null))
+					.withFunding(FUNDING_LIST)
+					.withFunding(null)
+					.withRelatedIdentifiers(RELATED_ID_LIST)
+					.withRelatedIdentifiers(ELRI)
+					.withVersion(VERSION_STRING_UNTRIMMED)
+					.build();
+			assertCreditMetadataFields(creditMetadata, IDENTIFIER_STRING, rt,
+					CONTRIBUTOR_LIST, TITLE_LIST, VERSION_MAP);
+		}
+	}
+
+	/**
+	 * Checks the error thrown when a CreditMetadata is built with invalid arguments
+	 *
+	 * @param builder
+	 *                CreditMetadata builder, complete with args
+	 * @param error
+	 *                the expected error
+	 */
+	private void buildCreditMetadataFailWithError(final CreditMetadata.Builder builder,
+			final String errorString) {
+		try {
+			builder.build();
+			fail(EXP_EXC);
+		} catch (Exception got) {
+			TestCommon.assertExceptionCorrect(got, new IllegalArgumentException(
+					"Errors in CreditMetadata construction:\n" + errorString));
+		}
+	}
+
+	@Test
+	public void buildFailInvalidPID() throws Exception {
+		for (final ResourceType rt : ResourceType.values()) {
```

Comment:
Again, if the purpose of the test and tests below is to check a specific error, I don't understand the benefit of cycling through the resource types

Reply:
N/A

Reply:
:+1: